pkg/interface/src/views/apps/chat/ChatResource.tsx

```diff
@@ -180,6 +179,7 @@ export function ChatResource(props: ChatResourceProps) {
           (!showBanner && hasLoadedAllowed) ? contacts : modifiedContacts
         }
         onUnmount={appendUnsent}
+        gcp={props.gcp}
```

Comment:
Instead of passing down gcp and s3 separately, please put them into the same prop and call it `storage`.

Reply:
Oh man, that is definitely sane and a mf'er to implement. On it...

Reply:
Sounds good.

Reply:
Hmm, I suppose this is most parsimoniously supported by starting from the store state itself – creating a top-level storage object and changing the reducers to match? Starting down that path assuming it is the way...

Reply:
Except all of the stores are changing (https://github.com/urbit/urbit/pull/4510) — I would continue to recommend making this PR as simple as possible and deploying it sooner than later to prevent colliding with that stuff.

Reply:
Done.

Reply:
Just merge — if you can push to get next-sys on master within the week, may be moot commentary.